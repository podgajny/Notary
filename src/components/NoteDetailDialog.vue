<script setup lang="ts">
import {
  Dialog,
  DialogContent,
  DialogHeader,
  DialogTitle,
} from "@/components/ui/dialog";

type Note = {
  id: string;
  title: string;
  body: string;
  createdAt: Date;
};

const props = defineProps<{
  note: Note;
  isOpen: boolean;
}>();

const emit = defineEmits<{
  (e: "update:isOpen", value: boolean): void;
}>();

// Formatuj datÄ™ do czytelnego formatu
const formatDate = (date: Date): string => {
  return new Intl.DateTimeFormat("en-US", {
    year: "numeric",
    month: "2-digit",
    day: "2-digit",
  }).format(date);
};

const handleOpenChange = (open: boolean) => {
  emit("update:isOpen", open);
};
</script>

<template>
  <Dialog :open="isOpen" @update:open="handleOpenChange">
    <DialogContent class="max-w-2xl p-6">
      <DialogHeader>
        <DialogTitle class="text-lg font-semibold text-slate-900">
          {{ note.title }}
        </DialogTitle>
      </DialogHeader>

      <div class="space-y-5">
        <div>
          <label class="block text-sm font-medium text-slate-700">Body</label>
          <div class="mt-1 whitespace-pre-wrap text-slate-900">
            {{ note.body }}
          </div>
        </div>

        <div>
          <p class="text-sm text-slate-600">
            <span class="font-medium">Created:</span>
            {{ formatDate(note.createdAt) }}
          </p>
        </div>
      </div>
    </DialogContent>
  </Dialog>
</template>
