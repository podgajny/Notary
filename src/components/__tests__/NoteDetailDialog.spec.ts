import { describe, it, expect, beforeEach, vi } from "vitest";
import { mount } from "@vue/test-utils";
import NoteDetailDialog from "../NoteDetailDialog.vue";

describe("NoteDetailDialog", () => {
  const mockNote = {
    id: "test-id-123",
    title: "Test Note Title",
    body: "This is the test note body content.\nWith multiple lines.",
    createdAt: new Date("2025-10-28T10:30:00.000Z"),
  };

  beforeEach(() => {
    vi.clearAllMocks();
  });

  it("powinien renderować tytuł notatki w DialogTitle", () => {
    // Arrange & Act
    const wrapper = mount(NoteDetailDialog, {
      props: {
        note: mockNote,
        isOpen: true,
      },
    });

    // Assert
    expect(wrapper.text()).toContain("Test Note Title");
  });

  it("powinien renderować treść notatki z zachowaniem formatowania", () => {
    // Arrange & Act
    const wrapper = mount(NoteDetailDialog, {
      props: {
        note: mockNote,
        isOpen: true,
      },
    });

    // Assert
    expect(wrapper.text()).toContain(
      "This is the test note body content.\nWith multiple lines.",
    );
  });

  it("powinien renderować datę utworzenia w formacie czytelnym dla użytkownika", () => {
    // Arrange & Act
    const wrapper = mount(NoteDetailDialog, {
      props: {
        note: mockNote,
        isOpen: true,
      },
    });

    // Assert - should display date in readable format
    expect(wrapper.text()).toContain("Created:");
    expect(wrapper.text()).toMatch(/\d{1,2}\/\d{1,2}\/\d{4}/); // Date pattern like MM/DD/YYYY or similar
  });

  it("powinien wyświetlić Dialog gdy isOpen jest true", () => {
    // Arrange & Act
    const wrapper = mount(NoteDetailDialog, {
      props: {
        note: mockNote,
        isOpen: true,
      },
    });

    // Assert
    expect(wrapper.find("[role='dialog']").exists()).toBe(true);
  });

  it("powinien ukryć Dialog gdy isOpen jest false", () => {
    // Arrange & Act
    const wrapper = mount(NoteDetailDialog, {
      props: {
        note: mockNote,
        isOpen: false,
      },
    });

    // Assert
    expect(wrapper.find("[role='dialog']").exists()).toBe(false);
  });

  it("powinien emitować zdarzenie 'update:isOpen' z wartością false gdy użytkownik zamyka Dialog", async () => {
    // Arrange
    const wrapper = mount(NoteDetailDialog, {
      props: {
        note: mockNote,
        isOpen: true,
      },
    });

    // Act - find and click close button
    const closeButton = wrapper.find("button[aria-label='Close']");
    await closeButton.trigger("click");

    // Assert
    expect(wrapper.emitted("update:isOpen")).toBeTruthy();
    expect(wrapper.emitted("update:isOpen")?.[0]).toEqual([false]);
  });

  it("powinien zawierać przycisk zamykający (X button)", () => {
    // Arrange & Act
    const wrapper = mount(NoteDetailDialog, {
      props: {
        note: mockNote,
        isOpen: true,
      },
    });

    // Assert - look for close button
    const closeButton = wrapper.find("button[aria-label='Close']");
    expect(closeButton.exists()).toBe(true);
  });

  it("powinien renderować poprawną strukturę komponentu", () => {
    // Arrange & Act
    const wrapper = mount(NoteDetailDialog, {
      props: {
        note: mockNote,
        isOpen: true,
      },
    });

    // Assert - snapshot test
    expect(wrapper.html()).toMatchSnapshot();
  });
});
