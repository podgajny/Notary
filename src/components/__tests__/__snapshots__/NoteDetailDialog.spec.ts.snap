// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NoteDetailDialog > powinien renderować poprawną strukturę komponentu 1`] = `
"<div class="flex flex-col gap-y-1.5 text-center sm:text-left"><h2 id="reka-dialog-title-v-0" class="tracking-tight text-lg font-semibold text-slate-900">Test Note Title</h2></div><div class="space-y-5"><div><label class="block text-sm font-medium text-slate-700">Body</label><div class="mt-1 whitespace-pre-wrap text-slate-900">This is the test note body content.
With multiple lines.</div></div><div><p class="text-sm text-slate-600"><span class="font-medium">Created:</span> 10/28/2025</p></div></div><button type="button" class="absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-accent data-[state=open]:text-muted-foreground"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-xicon w-4 h-4"><path d="M18 6 6 18"></path><path d="m6 6 12 12"></path></svg><span class="sr-only">Close</span></button>"
`;
